kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -869,6 +873,16 @@ func (c *ipamController) allocationIsValid(a *allocation, preferCache bool) bool
 }
 
 func (c *ipamController) syncIPAM() error {
+	ctx := context.Background()
```

Comment:
Hm, the only problem I see with this is that we'll now be calling `Get()` every time syncIPAM is run, which is pretty often. This will cause a lot of GET requests against the Kubernetes API server.

I think our options are:
- Use watch instead, and cache the latest clusterinformation 
- Move this Get() call further down, so we only check it when there's actually IPs / blocks to release. this means that we'll reduce the number of times we call Get() by taking it out of the mainline codepath for syncIPAM

My preference for this would be option 1 - use a watch on ClusterInformation instead. It should be relatively straightforward to add this to the controller's syncer (like we already do with the IPAM resources) and then add a new branch to `onUpdate` to handle updates to cluster information. Basically we just want to cache the latest update and then we can check the update here.